extras/bmp280/bmp280.c

```diff
@@ -46,116 +46,155 @@
 #define BMP280_REG_CONFIG      0xF5 /* bits: 7-5 t_sb; 4-2 filter; 0 spi3w_en */
 #define BMP280_REG_CTRL        0xF4 /* bits: 7-5 osrs_t; 4-2 osrs_p; 1-0 mode */
 #define BMP280_REG_STATUS      0xF3 /* bits: 3 measuring; 0 im_update */
+#define BMP280_REG_CTRL_HUM    0xF2 /* bits: 2-0 osrs_h; */
 #define BMP280_REG_RESET       0xE0
 #define BMP280_REG_ID          0xD0
 #define BMP280_REG_CALIB       0x88
+#define BMP280_REG_HUM_CALIB   0x88
 
 
-#define BMP280_CHIP_ID         0x58 /* BMP280 has chip-id 0x58 */
 #define BMP280_RESET_VALUE     0xB6
 
-typedef struct __attribute__((packed)) {
-    uint16_t dig_T1;
-    int16_t  dig_T2;
-    int16_t  dig_T3;
-    uint16_t dig_P1;
-    int16_t  dig_P2;
-    int16_t  dig_P3;
-    int16_t  dig_P4;
-    int16_t  dig_P5;
-    int16_t  dig_P6;
-    int16_t  dig_P7;
-    int16_t  dig_P8;
-    int16_t  dig_P9;
-} BMP280_Calib;
-
-static BMP280_Calib calib_data;
 
 void bmp280_init_default_params(bmp280_params_t *params)
 {
     params->mode = BMP280_MODE_NORMAL;
     params->filter = BMP280_FILTER_OFF;
     params->oversampling = BMP280_STANDARD;
+    params->oversampling_humidity = BMP280_STANDARD;
     params->standby = BMP280_STANDBY_250;
 }
 
-static uint8_t read_register8(uint8_t addr)
+static bool read_register8(uint8_t i2c_addr, uint8_t addr, uint8_t *value)
```

Comment:
After the changes this function seems unnecessary... or could be marked **inline**.